src/main/java/com/conveyal/datatools/manager/utils/MergeFeedUtils.java

```diff
@@ -0,0 +1,144 @@
+package com.conveyal.datatools.manager.utils;
+
+import com.conveyal.datatools.manager.jobs.FeedToMerge;
+import com.conveyal.datatools.manager.jobs.MergeStrategy;
+import com.conveyal.datatools.manager.models.FeedVersion;
+import com.conveyal.gtfs.error.NewGTFSError;
+import com.conveyal.gtfs.error.NewGTFSErrorType;
+import com.conveyal.gtfs.loader.Field;
+import com.conveyal.gtfs.loader.Table;
+import com.csvreader.CsvReader;
+import com.google.common.collect.Sets;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.zip.ZipFile;
+
+import static com.conveyal.gtfs.loader.Field.getFieldIndex;
+
+public class MergeFeedUtils {
+    private static final Logger LOG = LoggerFactory.getLogger(MergeFeedUtils.class);
+
+    /**
+     * Get the ids (e.g., trip_id, service_id) for the provided table from the zipfile.
+     */
+    public static Set<String> getIdsForTable(ZipFile zipFile, Table table) throws IOException {
+        Set<String> ids = new HashSet<>();
+        String keyField = table.getKeyFieldName();
+        CsvReader csvReader = table.getCsvReader(zipFile, null);
+        if (csvReader == null) {
+            LOG.warn("Table {} not found in zip file: {}", table.name, zipFile.getName());
+            return ids;
+        }
+        Field[] fieldsFoundInZip = table.getFieldsFromFieldHeaders(csvReader.getHeaders(), null);
+        // Get the key field (id value) for each row.
+        int keyFieldIndex = getFieldIndex(fieldsFoundInZip, keyField);
+        while (csvReader.readRecord()) ids.add(csvReader.get(keyFieldIndex));
+        csvReader.close();
```

Comment:
Please ensure this CsvReader is always closed. Maybe use a [try-with-resources block](https://stackoverflow.com/questions/54888904/should-i-always-close-bufferedreader/54889016#54889016).

Reply:
I don't think try with resources can be used because the class does not implement AutoCloseable. Are there scenarios you've seen where this is not getting closed?

Reply:
If there's any exception that occurs before the reader is closed, then the reader will be left open.

Reply:
Looks like this still needs to be implemented.

Reply:
Handled in https://github.com/ibi-group/datatools-server/pull/376/commits/06b2156db917fa518ba09f157a55369277cc7aa7.